services/web/server/src/simcore_service_webserver/resource_manager/garbage_collector.py

```diff
@@ -90,10 +97,56 @@ async def collect_garbage(registry: RedisResourceRegistry, app: web.Application)
                     user_id=int(dead_key["user_id"]),
                 )
 
+    # try to remove users which were marked as GUESTS manually
+    await remove_users_manually_marked_as_guests(
+        app=app, alive_keys=alive_keys, dead_keys=dead_keys
+    )
+
     # remove possible pending contianers
     await remove_orphaned_services(registry, app)
 
 
+async def remove_users_manually_marked_as_guests(
+    app: web.Application,
+    alive_keys: List[Dict[str, str]],
+    dead_keys: List[Dict[str, str]],
+) -> None:
+    """
+    Removes all the projects associated with GUEST users in the system.
+    If the user defined a TEMPLATE, this one also gets removed.
+    """
+
+    user_ids_to_ignore = set()
+    for entry in chain(alive_keys, dead_keys):
+        user_ids_to_ignore.add(int(entry["user_id"]))
+
+    guest_user_ids = await get_guest_user_ids(app)
+    logger.info("GUEST user id candidates to clean %s", guest_user_ids)
+
+    for guest_user_id in guest_user_ids:
+        if guest_user_id in user_ids_to_ignore:
+            logger.info(
+                "Ignoring user '%s' as it previously had alive or dead resource keys ",
+                guest_user_id,
+            )
+            continue
+        logger.info("Will try to remove resources for guest '%s'", guest_user_id)
+        # get all projects for this user and then remove with remove_resources_if_guest_user
```

Comment:
To access the app's ``ProjectDBAPI`` instance we normally access it to the instanced pre-initialized in ``app[APP_PROJECT_DBAPI]``

What you do works as well (with the current version of the code) and actually will even work even if the app was not correctly initialize but it goes against the design rationale, i.e. if somebody changes how ``PorjectDBAPI`` inits it might probably fail. 
In any case, this shows that the design is still not water-tight